src/java/voldemort/server/protocol/admin/AdminServiceRequestHandler.java

```diff
@@ -1131,12 +1135,18 @@ public void operate() {
 
                             String destinationDir = store.getStoreDirPath() + File.separator + "version-"
                                             + Long.toString(pushVersion);
+
+                            store.setCurrentlyFetching(true);
```

Comment:
Since you use an ```AtomicBoolean```, you might want to look into doing a compare-and-swap operation.